lib/validation_profiler/rules/list_validation_rule.rb

```diff
@@ -1,50 +1,48 @@
 module ValidationProfiler
   module Rules
+    # ListValidationRule class
     class ListValidationRule < ValidationProfiler::Rules::ValidationRule
-
       def error_message(field, attributes = {}, parent = nil)
-
         field_name = field.to_s
-        if parent != nil
-          field_name = "#{parent.to_s}.#{field.to_s}"
-        end
+        field_name = "#{parent}.#{field}" unless parent.nil?
 
-        #check if a custom error message has been specified in the attributes
-        if attributes[:message] == nil
-          #no custom error message has been specified so create the default message.
+        if attributes[:message].nil?
           "#{field_name} is not an accepted value."
         else
           attributes[:message]
         end
       end
 
-
       def validate(obj, field, attributes = {}, parent = nil)
-
-        #attempt to get the field value from the object
         field_value = get_field_value(obj, field)
 
-        if !is_required?(field_value, attributes)
-          return true
-        end
-
-        if field_value == nil
-          return false
-        end
+        return true unless is_required?(field_value, attributes)
+        return false if field_value.nil?
 
         list = attributes[:list]
-        if list == nil || !list.is_a?(Array)
-          raise ValidationProfiler::Exceptions::InvalidValidationRuleAttributes.new(ValidationProfiler::Rules::ListValidationRule, field)
-        end
+        regex = attributes[:regex]
 
-        if !list.include?(field_value)
-          return false
+        if regex.nil? && list.nil?
+          raise ValidationProfiler::Exceptions::InvalidValidationRuleAttributes
```

Comment:
Could move this call into a method so we can re-use it in the second location below:

```ruby
def raise_invalid_exception!(field)
  raise ValidationProfiler::Exceptions::InvalidValidationRuleAttributes
    .new(ValidationProfiler::Rules::ListValidationRule, field)
end
```

then your code becomes:

```ruby
if regex.nil? && list.nil?
  raise_invalid_exception!(field)
end
```

and

```ruby
if list
  unless list.is_a?(Array)
    raise_invalid_exception!(field)
```